src/hooks/useRequestProcessor.js

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react'
+import { useHeartbeat } from './useCourtContracts'
+import { useCourtClock } from '../providers/CourtClock'
+import { useRequestQueue } from '../providers/RequestQueue'
+import { useActivity } from '../components/Activity/ActivityProvider'
+
+import radspec from '../radspec'
+import allowedTermsBehind from '../allowedTermsBehind'
+
+export function useRequestProcessor() {
+  const { request } = useHeartbeat()
+  const { addActivity } = useActivity()
+  const { addRequests } = useRequestQueue()
+  const { neededTransitions } = useCourtClock()
+
+  const processRequests = useCallback(
+    requests => {
+      const processedRequests = []
+        .concat(requests)
+        .map(
+          ({
+            action,
```

Comment:
Maybe we should rename this key to `request`? Since the higher-level concept we're using is "requests".

Reply:
Makes sense

Reply:
The reason i used `action` is because the entire object is actually the request and we would have the problem where we would have to rename this as well https://github.com/aragon/court-dashboard/blob/0ec4b75a73f01a5c544c51cdbe6902f5a829d6d3/src/hooks/useRequestProcessor.js#L28